src/thor/triplegbuilder.cc

```diff
@@ -1627,6 +1630,18 @@ void TripLegBuilder::Build(
     trip_edge->set_source_along_edge(trim_start_pct);
     trip_edge->set_target_along_edge(trim_end_pct);
 
+    // We are looping the intermediate locs on the edge and if the edge id matches the loc edge id,
```

Comment:
this wont work when there are loops in the route. this is why i was suggesting that you keep an iterator to the current "through" location outside of the for loop and increment it each time a cut is performed. its both faster to do it this way and its robust to routes with loops in them